src/main/java/org/quantumbadger/redreader/views/LinkDetailsView.java

```diff
@@ -62,8 +62,13 @@ public LinkDetailsView(Context context, final String title, final String subtitl
 		layout.addView(textLayout);
 		((LinearLayout.LayoutParams)textLayout.getLayoutParams()).setMargins(0, marginPx, marginPx, marginPx);
 
-		final float linkFontScale = PrefsUtility.appearance_fontscale_comments(context, PreferenceManager.getDefaultSharedPreferences(context));
-		
+		final float linkFontScale;
+		if(PrefsUtility.appearance_fontscale_linkbuttons(context, PreferenceManager.getDefaultSharedPreferences(context)) != -1) {
```

Comment:
It might be better to put this logic instead `appearance_fontscale_linkbuttons()` (and similar methods), because at the moment these if statements are scattered across the code, and it's good to avoid methods returning magic numbers (-1).